services/port_manager/src/main/java/com/futurewei/alcor/portmanager/service/implement/PortServiceImpl.java

```diff
@@ -267,84 +372,70 @@ public PortStateJson updatePortState(String projectId, String portId, PortStateJ
         AsyncExecutor executor = new AsyncExecutor();
 
         PortState portState = portStateJson.getPortState();
-        PortState oldPortState = portRepository.findItem(portId);
+        portState.setProjectId(projectId);
 
         try {
-            if (portRepository.findItem(portId) == null) {
+            PortState oldPortState = portRepository.findItem(portId);
+            if (oldPortState == null) {
                 throw new PortStateNotFoundException();
             }
 
-            portState.setProjectId(projectId);
-
-            //Update mac_address
+            updatePortStateAsync(portState, oldPortState, executor, rollbacks);
 
-            //Update device_owner and device_id
-            String deviceOwnerNew = portState.getDeviceOwner();
-            String deviceIdNew = portState.getDeviceId();
-            String deviceIdOld = oldPortState.getDeviceId();
-            String tenantId = oldPortState.getTenantId();
-
-            if (deviceOwnerNew != null && deviceIdNew != null && !deviceIdNew.equals(deviceIdOld)) {
-                if (DeviceOwner.ROUTER.getOwner().equals(deviceOwnerNew)) {
-                    verifyRouter(deviceIdNew, tenantId);
-                }
-            }
+            //Wait for all async functions to finish
+            executor.joinAll();
 
-            //Update fixed_ips
-            List<PortState.FixedIp> fixedIps = portState.getFixedIps();
-            IpManagerProxy ipManagerProxy = new IpManagerProxy(rollbacks, projectId);
+            //Persist the new configuration of port to the db
+            portRepository.addItem(oldPortState);
```

Comment:
newPortState?

Reply:
no, newPortState not have complete fields.

Reply:
Sorry I don't quite follow this. Why do we want to persist the "oldPortState" in an update call? Can yo u please elaborate?

Reply:
Because some fields in PortState are generated dynamically during create, in addition, only part of the fields are allowed to be updated during update, so the fields in portState may be incomplete.